tests/kafkatest/tests/core/transactions_test.py

```diff
@@ -47,7 +47,10 @@ def __init__(self, test_context):
         self.num_output_partitions = 3
         self.num_seed_messages = 100000
         self.transaction_size = 750
-        self.transaction_timeout = 40000
+        # This is reducing the transaction cleanup interval.
+        # IN hard_bounce mode, transaction is broke ungracefully. Hence, it produces unstable
+        # offsets which obstructs TransactionalMessageCopier from receiving position of group.
+        self.transaction_timeout = 5000
```

Comment:
Hmm, still not sure about this. In a hard bounce, the broker failure will be detected after ZK session expiration, with a default of 18 sec. After that, new leaders will be elected and the client should recover. So, the transactional producer should be unblocked after 18 secs, even with a 40 sec transaction_timeout?

Reply:
> the broker failure will be detected after ZK session expiration

What is ungracefully stopped is client (TransactionMessageCopier) rather than broker in this issue. The transactional message sent by client is not correctly completed so there are pending transaction (unstable offset) stored by broker. When the client is restarting, it fails to get offsets of partition due to unstable offset (TransactionMessageCopier needs to get position of partition to calculate remaining messages after restarting).

The timeout of transaction is higher than the timeout of verification so this test fails if any transaction is not correctly abort.

Reply:
> we can adjust the comment to make it clear that the hard bounce is for the client.

done